modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectFieldResourceImpl.java

```diff
@@ -174,6 +180,18 @@ public ObjectField putObjectField(
 			Long objectFieldId, ObjectField objectField)
 		throws Exception {
 
+		com.liferay.object.model.ObjectField serviceBuilderObjectField =
+			_objectFieldService.getObjectField(objectFieldId);
+
+		com.liferay.object.model.ObjectDefinition
+			serviceBuilderobjectDefinition =
+				_objectDefinitionLocalService.getObjectDefinition(
+					serviceBuilderObjectField.getObjectDefinitionId());
+
+		if (!serviceBuilderobjectDefinition.isApproved()) {
+			_validateAndCreateListTypeDefinition(objectField);
```

Comment:
@murilostodolni please change to `_createListTypeDefinition` after moving the validation to the `localService`